privacyidea/lib/smsprovider/SMSProvider.py

```diff
@@ -211,6 +213,8 @@ def get_smsgateway(identifier=None, id=None):
             sqlquery = sqlquery.filter_by(id=id)
         except Exception:
             log.info("We can not filter for smsgateway {0!s}".format(id))
+    if gwtype:
+        sqlquery = sqlquery.filter_by(providermodule=gwtype)
```

Comment:
I haven't tested it, but if Firebase gateways are implemented as SMS gateways, don't we need to ensure that they do not turn up as SMS gateways at other places? e.g. in ``SmsTokenClass.get_class_info``.

Reply:
Hm, this is the question. Maybe not necessarily. It could be, that you also could use Firebase to send notifications  - like SMS. (with some slight adjustments)